zbus/src/connection/socket/unix.rs

```diff
@@ -348,7 +349,21 @@ fn get_unix_peer_creds_blocking(fd: RawFd) -> io::Result<crate::fdo::ConnectionC
             )
         };
 
+        let mut groups = User::from_uid(uid)?
+            .map(|user| CString::new(user.name))
+            .transpose()?
+            .map(|user| getgrouplist(&user, gid))
+            .transpose()
+            .ok()
+            .flatten()
+            .unwrap_or(Vec::new());
+        groups.sort_by_key(|gid| gid.as_raw());
+
         creds = creds.set_unix_user_id(uid.as_raw()).set_process_id(pid);
+
+        for group in groups.iter() {
+            creds = creds.add_unix_group_id(group.as_raw());
+        }
```

Comment:
probably more efficient to just do this as part of the chain to fetch the groups: `.for_reach(|g| creds.add_unix_group_id(g))` instead of creating a temporary array first. That is, if sorting is not needed.